src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -134,49 +91,71 @@ def test_WINNF_FT_S_FAD_1(self):
-                if activity_dump_file['recordType'] == 'esc_sensor':
-                    self.assertEqual(1, len(data['recordData']))
-                    # Verify the response file of Esc Sensor
-                    self.assertContainsRequiredFields("EscSensorRecord.schema.json", data['recordData'][0])
-                    self.assertDictEqual(esc_sensor_record, data['recordData'][0])
-                if activity_dump_file['recordType'] == 'zone':
-                    self.assertEqual(1, len(data['recordData']))
-                    # Verify the response file of PPA
-                    self.assertContainsRequiredFields("zoneData.schema.json", data['recordData'][0])
-                    self.assertEqual(ppa_record['id'], data['recordData'][0]['id'])
-                    self.assertEqual(ppa_record['usage'], data['recordData'][0]['usage'])
-                    self.assertEqual(ppa_record['terminated'], data['recordData'][0]['terminated'])
-                    self.assertEqual(ppa_record['zone']['features'][0]['geometry']['coordinates'],\
-                                      data['recordData'][0]['zone']['features'][0]['geometry']['coordinates'])
-                    self.assertEqual(set(ppa_record['ppaInfo']['palId']),\
-                                      set(data['recordData'][0]['ppaInfo']['palId']))
-                    self.assertEqual(set(ppa_record['ppaInfo']['cbsdReferenceId']), \
-                                     set(data['recordData'][0]['ppaInfo']['cbsdReferenceId']))
-                    self.assertEqual(ppa_record['ppaInfo']['ppaBeginDate'],\
-                                      data['recordData'][0]['ppaInfo']['ppaBeginDate'])
-                    self.assertEqual(ppa_record['ppaInfo']['ppaExpirationDate'],\
-                                      data['recordData'][0]['ppaInfo']['ppaExpirationDate'])                                    
\ No newline at end of file
+        for activity_dump_file in response['files']:
+            # Verify the record type
+            # Verify the response files with ActivityDumpFile.schema.json Object schema
+            self.assertContainsRequiredFields("ActivityDumpFile.schema.json", activity_dump_file)
+            # Get json data from url
+            data = self._sas.DownloadFile(activity_dump_file['url'])
+            if activity_dump_file['recordType'] == 'cbsd':
+                 # Verify that everything in the full dump matches a cbsd or grant created at the beginning
+                for record in data['recordData']:
+                # Verify the response files with CbsdData.schema.json Object schema
+                    self.assertContainsRequiredFields("CbsdData.schema.json", record)
+                    self.assertTrue(record['registration']['fccId'] in (device_a['fccId'], device_c['fccId']))
+                # Verify all the previous activities on CBSDs and Grants exist in the dump files    
+                for index, device in enumerate([device_a, device_c]):
```

Comment:
If I included the data for device_a twice, would this code catch it?

Reply:
Please reconsider this comment.

Reply:
I think it will catch in the next code :
 34. cbsd_record = [record['registration'] for record in recordData if record['id'] == record_id]
 35. self.assertEqual(1, len(cbsd_record))

right?